libp2p/crypto/chacha20poly1305.nim

```diff
@@ -28,6 +28,10 @@ type
   ChaChaPolyNonce* = array[ChaChaPolyNonceSize, byte]
   ChaChaPolyTag* = array[ChaChaPolyTagSize, byte]
 
+proc intoChaChaPolyKey*(s: array[32, byte]): ChaChaPolyKey =
+  assert s.len == ChaChaPolyKeySize
```

Comment:
wait, why is the `copyMem` needed here? it's the same type?

to simplify, `intoChaChaPolyKey(s: openArray[byte])` would cover both `seq` and `array` - even though it would introduce an unnecessary runtime check in the latter case, I don't think it actually matters, so I'd start there.

Reply:
Oh yeah that was a `converter` actually, that's why it has this signature.
But we decided (wisely) to not use converters. I will fix that bit, can remove the rest too actually (`seq`)
Nice catch!